core/variant.cpp

```diff
@@ -1662,7 +1662,17 @@ Variant::operator Transform() const {
 		return Transform(*_data._basis, Vector3());
 	else if (type == QUAT)
 		return Transform(Basis(*reinterpret_cast<const Quat *>(_data._mem)), Vector3());
-	else
+	else if (type == TRANSFORM2D) {
+		const Transform2D &t = *_data._transform2d;
+		Transform m;
+		m.basis.elements[0][0] = t.elements[0][0];
+		m.basis.elements[1][0] = t.elements[1][0];
```

Comment:
on the C++ side, colums/rows are swapped in Transform vs Transform 2D. Transform uses them this way for performance (better autovectorizing).

Reply:
Right, I saw that before implementing this and then messed up the actual implementation. Ill submit a new PR.